src/function/cast_string_non_nested_functions.cpp

```diff
@@ -88,31 +88,58 @@ bool TryCastStringToTimestamp::tryCast<timestamp_sec_t>(const char* input, uint6
     return true;
 }
 
-static bool isDate(const std::string& str) {
-    return RE2::FullMatch(str, "\\d{4}/\\d{1,2}/\\d{1,2}") ||
-           RE2::FullMatch(str, "\\d{4}-\\d{1,2}-\\d{1,2}") ||
-           RE2::FullMatch(str, "\\d{4} \\d{1,2} \\d{1,2}") ||
-           RE2::FullMatch(str, "\\d{4}\\\\\\d{1,2}\\\\\\d{1,2}");
+static RE2& datePattern1() {
```

Comment:
Move regular expressions for date patterns inside date.h. and define them as static constexpr.

Reply:
RE2 is not a literal type, so you can't declare it as a constexpr. It should be fine to just evaluate it once at runtime